docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+Generally speaking the inputs required for the aforementioned results tend to be
+the molecule, level of theory (method plus basis set), and method parameters.
+While the set of potential parameters is huge, again there are only a handful of
+commonly used parameters.
+
+Primary targets are: algorithm variations (e.g. density fitting), number of
+iterations, and convergence criteria.
+
+Common Quantum Chemistry Calculations
+=======================================
+As described in the design principles, the UI should enable users to conduct
+commonly used quantum chemistry calculations such as single-point energies and
+geometry optimizations in as few steps as possible. It should also be flexible
+enough to allow for more complex workflows using the helper/driver functions and
+PluginPlay modules. Users can execute these calculations either via Python
+scripts or interactively using Jupyter notebooks. For those who desire a
+fine-grained control of their workflow, they can utilize the Python and/or C++
+API.
+
+1. **Single Point Energy**
+   
+   In these calculations total energy is computed for a fixed geometry of the
+   molecule (arrangement of atomic nuclei) corresponding to a single point in
+   the potential energy surface. While energy calculation is the most common
+   case, the user might be interested in obtaining the wave function,
+   derivatives, or other properties (dipole moment, electron density, etc.) for
+   this specific geometry. 
+
+2. **Geometry Optimization**
+   
+   This is a procedure to find the arrangement of atomic nuclei that corresponds
+   to a stationary point in the potential energy surface. This procedure
+   generally requires the calculation of the gradients and Hessian (first and
+   second order partial derivatives of the energy with respect to nuclear
+   coordinates, respectively) at many different geometries. 
+
+3. **Vibrational Frequency Calculation**
+   
+   These calculations provide the vibrational modes of a molecule and their
+   corresponding frequencies. The procedure requires Hessian calculation at the
+   equilibrium  geometry.
+
+4. **Molecular Dynamics Simulation**
+   
+   These calculations allow for the study of the trajectory of atoms and
+   molecules over time using classical dynamics. Gradient calculations are
+   required to compute the forces acting on the atoms.
+
+We will initially focus on the UX design of the "single point energy"
+calculations in this document since other types of calculations (geometry
+optimization, vibrational frequency calculation, etc.) can be performed through
+external codes rather easily as long as the UX for "single point calculations"
+provides the appropriate support. 
+
+Before we delve into the specifics of the NWChemEx UX, we provide examples of
+SCF/sto-3g energy calculations for a hydrogen molecule using PySCF, PSI4, and
+MolSSI QCEngine. Finally, we will provide an NWChemEx example and discuss about
+the choices made.
+
+PySCF
```

Comment:
I'd demote this to a subsection of `Common Quantum Chemistry Calculations` or create a new section `Existing UIs` and have these be subsections of that section. Also, your overlining/underlining is not consistent with our [conventions](https://nwchemex-project.github.io/.github/conventions/rst_conventions.html) (which in turn made it difficult for me to assess the nesting of these sections upon inspection).

Reply:
Oops, fixed with c28a7cb41f1d7d9cb6d97078ae526af84a8eb86f.